src/compas_timber/_fabrication/step_joint_notch.py

```diff
@@ -0,0 +1,844 @@
+        elif step_depth == 0.0 and heel_depth > 0.0:
+            if tapered_heel:
+                return StepShape.TAPERED_HEEL
+            else:
+                return StepShape.HEEL
+        elif step_depth > 0.0 and heel_depth > 0.0:
+            return StepShape.DOUBLE
+        else:
+            raise ValueError("At least one of step_depth or heel_depth must be greater than 0.0.")
+
+    @staticmethod
+    def _calculate_displacement_end(strut_height, strut_inclination, orientation):
+        # Calculates the linear displacement from the origin point to the end of the notch based on the strut_height and strut_inclination.
+        displacement_end = strut_height / math.sin(math.radians(strut_inclination))
+        if orientation == OrientationType.END:
+            displacement_end = -displacement_end  # negative displacement for the end cut
+        return displacement_end
+
+    @staticmethod
+    def _calculate_displacement_heel(heel_depth, strut_inclination, orientation):
+        # Calculates the linear displacement from the origin point to the heel of the notch based on the heel_depth and strut_inclination.
+        displacement_heel = abs(
+            heel_depth / (math.sin(math.radians(strut_inclination)) * math.cos(math.radians(strut_inclination)))
+        )
+        if orientation == OrientationType.END:
+            displacement_heel = -displacement_heel
+        return displacement_heel
+
+    ########################################################################
+    # Methods
+    ########################################################################
+
+    def apply(self, geometry, beam):
+        """Apply the feature to the beam geometry.
+
+        Parameters
+        ----------
+        geometry : :class:`~compas.geometry.Brep`
+            The beam geometry to be milled.
+        beam : :class:`compas_timber.elements.Beam`
+            The beam that is milled by this instance.
+
+        Raises
+        ------
+        :class:`~compas_timber.elements.FeatureApplicationError`
+            If the cutting planes do not create a volume that itersects with beam geometry or any step fails.
+
+        Returns
+        -------
+        :class:`~compas.geometry.Brep`
+            The resulting geometry after processing
+
+        """
+        # type: (Brep, Beam) -> Brep
+        # get cutting planes from params
+        try:
+            cutting_planes = self.planes_from_params_and_beam(beam)
+        except ValueError as e:
+            raise FeatureApplicationError(
+                None, geometry, "Failed to generate cutting planes from parameters and beam: {}".format(str(e))
+            )
+        # create notch polyedron from planes
+        # add ref_side plane to create a polyhedron
+        # !: the beam's ref_side Plane might need to be offsetted to create a valid polyhedron when step_type is "double"
+        cutting_planes.append(Plane.from_frame(beam.ref_sides[self.ref_side_index]))
```

Comment:
![image](https://github.com/user-attachments/assets/c4d92beb-c201-4f80-a507-b6fc1179655f) 
![image](https://github.com/user-attachments/assets/b857c6c5-29cc-4fc9-abe0-ec74233ebdd6)

@chenkasirer I guess this is going to be an issue if I want to use rhino 7, right?